zerver/tests/test_signup.py

```diff
@@ -3899,7 +3899,6 @@ def test_do_not_deactivate_final_admin(self) -> None:
 
     def test_do_not_deactivate_final_user(self) -> None:
         realm = get_realm('zulip')
-        do_change_user_role(self.example_user("desdemona"), UserProfile.ROLE_MEMBER)
```

Comment:
Why did you remove this?  I think the actual change we want to make is that we should have a prep commit replacing any queries filtering on `role=UserProfile.ROLE_REALM_ADMINISTRATOR` to use a `__in` clause in the few places that appears in the codebase.

Reply:
This was added previously so that we do not have to remove `Desdemona` separately as we wanted to come to a stage with last user in realm.

I actually intended to change the query to exclude realm owner only and keep realm owner as the last user to check deactivating of final user and I missed to change that.

Reply:
Yeah, I think ultimately we want this test to confirm that the last owner can't be removed.

Reply:
To be clear, this test is to confirm that last user cannot be deactivated, the test for last owner is separte.